test/unit/sdam/monitor.test.ts

```diff
@@ -306,6 +307,53 @@ describe('monitoring', function () {
         }, minHeartbeatFrequencyMS);
       });
     });
+
+    describe('roundTripTime', function () {
+      const table = [
+        {
+          serverMonitoringMode: 'stream',
+          topologyVersion: {
+            processId: new ObjectId(),
+            counter: new Long(0, 0)
+          }
+        },
+        { serverMonitoringMode: 'poll', topologyVersion: undefined }
+      ];
+      for (const { serverMonitoringMode, topologyVersion } of table) {
+        context(`when serverMonitoringMode = ${serverMonitoringMode}`, () => {
+          context('when more than one heartbeatSucceededEvent has been captured', () => {
+            let heartbeatDurationMS = 100;
+            it('correctly returns the mean of the heartbeat durations', async () => {
+              mockServer.setMessageHandler(request => {
+                setTimeout(
+                  () => request.reply(Object.assign({ helloOk: true }, mock.HELLO)),
+                  heartbeatDurationMS
+                );
+                heartbeatDurationMS += 100;
+              });
+              const server = new MockServer(mockServer.address());
+              if (topologyVersion) server.description.topologyVersion = topologyVersion;
+              monitor = new Monitor(server as any, { serverMonitoringMode } as any);
+              monitor.connect();
+
+              for (let i = 0; i < 5; i++) {
+                await once(monitor, 'serverHeartbeatSucceeded');
+                monitor.requestCheck();
+              }
+
+              const avgRtt = monitor.roundTripTime;
+              const expectedRtt = 300; // (100 + 200 + 300 + 400 + 500)/5 = 300
+              // avgRtt will strictly be greater than expectedRtt since setTimeout sets a minimum
+              // delay from the time of scheduling to the time of callback execution
+              expect(avgRtt - expectedRtt).greaterThanOrEqual(0);
+              expect(avgRtt - expectedRtt).lessThan(50);
```

Comment:
Can we express this as a `within` chai assertion? And is +/-50ms the best we can do?

Reply:
Will refactor to use the `within` assertion

Running locally, I've seen up to 25ms discrepancy overall, so I doubled that to account for any differences in my hardware and running in CI.

Reply:
Fine by me then, that range isn't critical and might as well keep the test far away from the flaky zone